### PR TITLE
Mobile: Add sync status icon

### PR DESCRIPTION
**Pull request by personalizedrefrigerator**
# Summary
 * [Feature discussion](https://discourse.joplinapp.org/t/mobile-display-sync-status-while-in-edit-view/27034/4)
 * Adds an icon button that shows whether changes made in the mobile editor haven't been synced to the server.

# Testing plan
1. In "configuration", disable syncing
2. Edit a note.
    - The "sync status" icon should be invisible
3. In "configuration", set a cloud service as the sync target
4. Open a note.
    - The sync icon should be a "push" icon before changes are made/have been pushed to the server 
5. Push the "sync" button
    - After syncing, the icon should contain a checkmark.
6. Edit the note
    - After making a change, but before syncing, the button should contain a "push icon" again 
7. Press the "sync" button
8. In "configuration", enable syncing with the filesystem, but with an empty/invalid output directory
9. Open a note and press the "sync" button.
    - After pressing the button once, the cloud should contain an exclamation point.
    - Attempting to sync again should display an error message.

# Screenshots

 - When up-to-date
     - Outlined cloud with a checkmark

 - While syncing
     - Outlined cloud with a 'syncing' symbol

 - When changes can be pushed
     - Outlined cloud with an upward-pointing arrow

 - After sync errors
     - Outlined cloud containing an exclamation point

### CODE REVIEW

File: packages/app-mobile/components/SyncStatusIcon.tsx [hunk 1]
```diff
@@ -0,0 +1,197 @@
...
+	const syncAnim = useRef(new Animated.Value(0)).current;
+
+	useEffect(() => {
+		if (props.syncStarted) {
+			Animated.loop(Animated.timing(syncAnim, {
+				toValue: 360,
+				duration: 5000,
+				easing: Easing.linear,
+				useNativeDriver: true,
+			})).start();
```
Comment (laurent22) [inline packages/app-mobile/components/SyncStatusIcon.tsx]: Does that get cleaned up when the component is unmounted?

**Comment by laurent22** [comment]
Could you add a few screenshots showing the feature please?

File: packages/app-mobile/components/SyncStatusIcon.tsx (hunk 1 above)
Comment (personalizedrefrigerator) [reply packages/app-mobile/components/SyncStatusIcon.tsx]: I don't think it was getting cleaned up. It should be fixed now.

**Comment by personalizedrefrigerator** [comment]
> Could you add a few screenshots showing the feature please?



https://user-images.githubusercontent.com/46334387/187069295-6fa5d1aa-a9f4-464e-9ade-7c890eb2d578.mp4

**Comment by narFnarF** [comment]
Thanks for the screenshots and video. It looks very good!

**Comment by laurent22** [comment]
Do you have examples of applications doing this? It feels a bit like clutter on the UI and many people won't know what these icons mean. Or perhaps the solution is just to trigger a sync very shortly after a note has been changed (now I think it's a few seconds), in which case such icons become less necessary.

**Comment by personalizedrefrigerator** [comment]
Closing for now. We can discuss alternatives on the forum.

**Comment by narFnarF** [comment]
I understand the need to keep clutter to a minimum. It's always a difficult balance between displaying too much information or not enough. In this case though, I think it's very important to display: It's about knowing if the work I just did will be preserved or not!

Unless you can guarantee that the last word I typed will be pushed to the server, even if I minimize the app by pressing the Home button on my iphone, we need to show sync status. As a user, it's super important for me to know that my work has been saved and synced.

@personalizedrefrigerator :
> Closing for now. We can discuss alternatives on the forum.

Have you created a thread for this? We should link it here.

**Comment by narFnarF** [comment]
# Google Doc
Cloud icon with a checkmark:


# Google Doc
When you click on the cloud icon:


# Nextcloud Notes
Small asterisk* next to the note in the side panel. The asterisk disappear when the note is synced. This is very barebone and hard to understand for non-technical users. The Google Doc way is better. But still, it's much better than nothing.


# Nextcloud markdown editor
Small floating text "Saving" in the top right corner. The text becomes "Saved" once the doc has been synced


# Collabora (Nextcloud Suite)
Click the checkmark button on the left (not super clear it's a Save button). The text in top center becomes "Last saved: X seconds/minutes ago"


# Wordpress app on iPhone
Display small text "Update" (in french: "mise à jour") near the title of the post. The text color change state based on if there are changes to push on the server or not. Tapping the text pushes the changes to the server.


# Need more?
I can provide more example if you need @laurent22 
If you think this is too big, the icon's size could be reduced (careful to keep it big enough to be comfortable to click of course!)

**Comment by narFnarF** [comment]
> Closing for now. We can discuss alternatives on the forum.

Have your created a thread for this?

**Comment by personalizedrefrigerator** [comment]
> Have you created a thread for this? We should link it here.

Let's continue using https://discourse.joplinapp.org/t/mobile-display-sync-status-while-in-edit-view/27034

**Pull request closed without merging**